sfge/src/main/java/com/salesforce/rules/dmlinloop/DmlInLoopRuleHandler.java

```diff
@@ -0,0 +1,72 @@
+package com.salesforce.rules.dmlinloop;
+
+import com.salesforce.exception.ProgrammingException;
+import com.salesforce.graph.ApexPath;
+import com.salesforce.graph.symbols.CloningSymbolProvider;
+import com.salesforce.graph.symbols.DefaultSymbolProviderVertexVisitor;
+import com.salesforce.graph.symbols.SymbolProvider;
+import com.salesforce.graph.vertex.*;
+import com.salesforce.graph.visitor.ApexPathWalker;
+import com.salesforce.rules.Violation;
+import java.util.Set;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+
+public class DmlInLoopRuleHandler {
+
+    /** checks if a certain vertex is of interest to this rule */
+    public boolean test(BaseSFVertex vertex) {
+        /* Three different types of DML statements we need to check for:
+           1. insert a;
+           2. [SELECT a FROM b WHERE c]
+           3. Database.something();
+        */
+        if (vertex instanceof DmlStatementVertex) {
+            return true;
+        } else if (vertex instanceof SoqlExpressionVertex) {
+            return true;
+        } else if (vertex instanceof MethodCallExpressionVertex) {
+            return true;
```

Comment:
For performance reasons, we should check whether the method is a `Database.<whatever>` method here, not in `DmlInLoopVisitor`. That way, we avoid considering vertices unnecessarily.